hazelcast/src/main/java/com/hazelcast/internal/partition/impl/PartitionStateManager.java

```diff
@@ -464,18 +428,13 @@ int replaceMember(Member oldMember, Member newMember) {
     }
 
     PartitionTableView getPartitionTable() {
-        int version = 0;
-        if (node.getClusterService().getClusterVersion().isUnknownOrLessOrEqual(Versions.V4_0)) {
-            //RU_COMPAT_4_0
-            version = getVersion();
-        }
-        return new PartitionTableView(getPartitionsCopy(true), version);
+        return new PartitionTableView(getPartitionsCopy(true));
     }
 
     @Override
     public void onClusterVersionChange(Version newVersion) {
         if (newVersion.isEqualTo(Versions.V4_1) && initialized) {
```

Comment:
could this version comparison be removed  also?--> `newVersion.isEqualTo(Versions.V4_1)`

Reply:
good catch, `PartitionStateManager` and `InternalPartitionServiceImpl` listened for cluster version changes in order to initialize partition versions on transition from 4.0 to 4.1 cluster version. This is no longer necessary as 4.2 members can only partitipate in 4.1 / 4.2 clusters -> partition version is already initialized before a 4.2 member can join the cluster.